### DISCUSSION TITLE
Setting state at app level from child is not working.

### DISCUSSION THREAD

**Discussion by ilabsllc** [vercel/next.js#37644] [Help]
In my custom App, I have defined the state, along with a callback to set the state.

```js
const [navItem, setNavItem] = useState("Home");

const setNavItemCallback = (item) => {
    // do something with value in parent component, like save to state
    console.log(`setting nav item in app to ${item}`)
    setNavItem(item);
  }
```

I  pass the callback in my App.

```html
<Layout config={UiConfig} navItem={navItem} setNavItem={setNavItemCallback}>
      <Component {...pageProps} key={router.asPath} />
    </Layout>
```

I set the state in my nav link whenever the user clicks on one of the navigation items.

```js
import { useRouter } from 'next/router'

function NavLink(props) {
  const router = useRouter()
  const { children, href, setNavItem } = props;

  const handleClick = (e) => {
    e.preventDefault()
    setNavItem(children);
    router.push(href);
  }

  return (
    <a  className={router.pathname === href ? "nav-link active" : "nav-link"}  onClick={handleClick}>
      {children}
    </a>
  )
}

export default NavLink
```

I verified the handleClick is getting called, and the children ( string in this case ) is correct. There is no error from setNavItem(children) call, but it doesn't seem to set the state to the new value.

What am I doing wrong?

**Comment by icyJoseph** [comment]
Hi,

This works without issue. Could you share a repository to take a look at? How's layout defined? What is it that you are expecting to see, but not seeing? I suspect a few things might be going on, but I don't want to confuse you.

A repository where this fails, would help us resolve this much faster.

**Reply by ilabsllc** [reply]
Thanks, @icyJoseph. Let me debug this some more then if you don't see anything obviously wrong. All I am attempting to do is detect the route change as the user clicks on nav bar links, and then change the props that get passed to the layout to display a different sidebar by passing a different list data or not show the sidebar at all in case of the landing page.

setNavItemCallback never gets called above.

**Reply by icyJoseph** [reply]
That's really strange. I did see the logs and all. Please consider sharing more of the application, I'm curious to see where things are going wrong.
